src/main/java/org/stroganov/dao/DAOFactory.java

```diff
@@ -17,6 +19,13 @@ public static LibraryDAO getLibraryDAO(DAOType type) throws DBExceptions {
             case JSON:
                 dao = JsonLibraryDAO.getInstance();
                 break;
+            case WEB_SERVICE_SOAP:
```

Comment:
Switch case is almost always bad decision.
When you add a new service , you will add a new switch , and if you add 5 more services , then there will be five more switches and so on indefinitely ... so the code will become very difficult to maintain
lets try to remove it

Reply:
Remove this class and use code in enum 
```suggestion
            type.getLibraryDAO();
```

Reply:
fixed - ea1f2878707fddfa58f63b714954fcac2810ea26